### PR TITLE
Fix discovering/running of unit tests due to file extension issue

### LINKED ISSUE #402: unit tests are not run from R CMD CHECK
From the `testthat` docs under the section [integration with r cmd check](https://github.com/hadley/testthat#integration-with-r-cmd-check) it says the test files can be .r or .R, but the main `tests/testthat.R` must be with a capital .R not .r.  
Currently we have testthat.r and it is not running on the build in the [check package travis step](https://github.com/IRkernel/IRkernel/blob/master/.travis.sh#L70).

### PR DESCRIPTION
**Pull request by poplav**
Fixes #402

### CODE REVIEW

**Comment by poplav** [comment]
Now that tests are enabled/running it looks like there is an error:

```
  testthat results ================================================================
  OK: 9 SKIPPED: 0 FAILED: 1
  1. Failure: kernel tests pass (@test_kernel.r#6) 
```

This is the integration level test.  I think the correct route would be to move this out of the testthat framework into it's own python call in [travis.sh - `test_kernel`](https://github.com/IRkernel/IRkernel/blob/master/.travis.sh#L86).  testthat `check_package` should be reserved for unit tests.

**Comment by poplav** [comment]
Can someone explain the log line count checking going on as a result of R CMD check.  I had to increase the count by one to get it to pass because of the error:

```
++grep -q WARNING /home/travis/build/IRkernel/IRkernel/IRkernel.Rcheck/00check.log
++grep -q 'Status..1 NOTE' /home/travis/build/IRkernel/IRkernel/IRkernel.Rcheck/00check.log
+++grep -v '* .*$' /home/travis/build/IRkernel/IRkernel/IRkernel.Rcheck/00check.log
+++wc -l
+++sed -e 's/^[[:space:]]*//'
++LINES=6
++echo 'Lines: 6'
Lines: 6
++[[ _6 != _5 ]]
++grep -v '* .*$' /home/travis/build/IRkernel/IRkernel/IRkernel.Rcheck/00check.log
Found the following calls to attach():
File ‘IRkernel/R/execution.r’:
  attach(NULL, name = "jupyter:irkernel")
See section ‘Good practice’ in ‘?attach’.
  Running ‘testthat.R’
Status: 1 NOTE
++false
```

File: .travis.sh [hunk 1]
```diff
@@ -80,7 +80,7 @@ check_package() (
     LINES=$(grep -v '* .*$' "$CHECK_LOG" | wc -l | sed -e 's/^[[:space:]]*//')
     echo "Lines: $LINES"
     #  4 lines from the "attach" NOTE, 1 lines from the "Status" at the end
-    if [[ "_$LINES" != _5 ]]; then grep -v '* .*$' "$CHECK_LOG"; false; fi
+    if [[ "_$LINES" != _6 ]]; then grep -v '* .*$' "$CHECK_LOG"; false; fi
```
Comment (flying-sheep) [inline .travis.sh]: the comment above still says “4 lines + 1 line”.
could you edit it to explain where the 6th comes from?

**Comment by flying-sheep** [comment]
i’m not sure why this should be necessary. see #402 for discussion

**Comment by flying-sheep** [comment]
closing, superseeded by #404

**Pull request closed without merging**